database/mysql/detectors-mysql.tf

```diff
@@ -0,0 +1,222 @@
+EOF
+
+  rule {
+    description           = "with rate > ${var.mysql_questions_anomaly_fire_growth_rate_threshold}"
+    severity              = "Critical"
+    detect_label          = "CRIT"
+    disabled              = coalesce(var.mysql_questions_anomaly_disabled, var.detectors_disabled)
+    notifications         = coalescelist(var.mysql_questions_anomaly_notifications, var.notifications)
+    parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{readableRule}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+  }
+
+}
+
+resource "signalfx_detector" "mysql_replication_lag" {
+  name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] MySQL replication lag"
+
+  program_text = <<-EOF
+    signal = data('mysql_seconds_behind_master', filter=${module.filter-tags.filter_custom}, rollup='average')${var.mysql_replication_lag_aggregation_function}${var.mysql_replication_lag_transformation_function}.publish('signal')
+    detect(when(signal > ${var.mysql_replication_lag_threshold_critical})).publish('CRIT')
+    detect(when(signal > ${var.mysql_replication_lag_threshold_warning}) and when(signal <= ${var.mysql_replication_lag_threshold_critical})).publish('WARN')
+  EOF
+
+  rule {
+    description           = "is too high > ${var.mysql_replication_lag_threshold_critical}"
+    severity              = "Critical"
+    detect_label          = "CRIT"
+    disabled              = coalesce(var.mysql_replication_lag_disabled_critical, var.mysql_replication_lag_disabled, var.detectors_disabled)
+    notifications         = coalescelist(var.mysql_replication_lag_notifications_critical, var.mysql_replication_lag_notifications, var.notifications)
+    parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{readableRule}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+  }
+
+  rule {
+    description           = "is too high > ${var.mysql_replication_lag_threshold_warning}"
+    severity              = "Warning"
+    detect_label          = "WARN"
+    disabled              = coalesce(var.mysql_replication_lag_disabled_warning, var.mysql_replication_lag_disabled, var.detectors_disabled)
+    notifications         = coalescelist(var.mysql_replication_lag_notifications_warning, var.mysql_replication_lag_notifications, var.notifications)
+    parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{readableRule}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+  }
+}
+
+resource "signalfx_detector" "mysql_replication_status" {
+  name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] MySQL replication status"
+
+  program_text = <<-EOF
+    signal = data('mysql_slave_sql_running', filter=${module.filter-tags.filter_custom}, rollup='average')${var.mysql_replication_status_aggregation_function}${var.mysql_replication_status_transformation_function}.publish('signal')
+    detect(when(signal < 1)).publish('CRIT')
+EOF
+
+  rule {
+    description           = "is stopped"
```

Comment:
`not running` seems more appropriate as the replication can be stopped manually or can also be broken.

Reply:
ok